heareval/predictions/task_predictions.py

```diff
@@ -91,28 +97,160 @@ def forward(self, x):
     def training_step(self, batch, batch_idx):
         # training_step defined the train loop.
         # It is independent of forward
-        x, y = batch
+        x, y, _ = batch
         y_hat = self.predictor.forward_logit(x)
         loss = self.predictor.logit_loss(y_hat, y)
         # Logging to TensorBoard by default
         self.log("train_loss", loss)
         return loss
 
     def validation_step(self, batch, batch_idx):
-        x, y = batch
+        # -> Dict[str, Union[torch.Tensor, List(str)]]:
+        x, y, metadata = batch
         y_hat = self.predictor.forward_logit(x)
-        loss = self.predictor.logit_loss(y_hat, y)
         y_pr = self.predictor(x)
-        # Logging to TensorBoard by default
-        self.log("val_loss", loss, prog_bar=True)
-        for score in self.scores:
-            self.log(f"val_{score}", score(y_pr, y), prog_bar=True)
+        z = {
+            "prediction": y_pr,
+            "prediction_logit": y_hat,
+            "target": y,
+        }
+        # https://stackoverflow.com/questions/38987/how-do-i-merge-two-dictionaries-in-a-single-expression-taking-union-of-dictiona
+        return {**z, **metadata}
+
+    # Implement this for each inheriting class
+    # TODO: Can we combine the boilerplate for both of these?
```

Comment:
I'm not sure if we can simplify the boilerplate in validation_epoch_end implementations further